src/test/java/com/example/weatherwaveapi/service/YandexWeatherServiceTest.java

```diff
@@ -0,0 +1,192 @@
+import com.example.weatherwaveapi.model.response.yandexapi.weather.LocalityInfo;
+import com.example.weatherwaveapi.model.response.yandexapi.weather.YandexWeatherApiResponse;
+import com.example.weatherwaveapi.model.response.yandexapi.weather.YandexWeatherResponse;
+import com.example.weatherwaveapi.util.urlbuilder.OpenWeatherUrlBuilder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.assertj.core.api.InstanceOfAssertFactories;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static com.example.util.WeatherApiUtil.*;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+
+public class YandexWeatherServiceTest {
+    private MockWebServer mockWebServer;
+    private GeneralSettings generalSettings;
+    private OpenWeatherUrlBuilder urlBuilder;
+    private YandexWeatherService yandexService;
+    private static final String LOCALHOST = "http://localhost:";
+    private static final String SKIP = "";
+    private static final String PLACEHOLDER = "%s";
+
+    @BeforeEach
+    void setup() throws IOException {
+        mockWebServer = new MockWebServer();
+        generalSettings = new GeneralSettings();
+        urlBuilder = new OpenWeatherUrlBuilder(generalSettings);
+        YandexApi yandexApi= new YandexApi(SKIP, LOCALHOST + mockWebServer.getPort(), SKIP);
+        generalSettings.setYandexApi(yandexApi);
+        yandexService = new YandexWeatherService(new RestTemplate(), generalSettings, urlBuilder);
+        String url = String.format(LOCALHOST+PLACEHOLDER, mockWebServer.getPort());
+        mockWebServer.url(url + "/weather");
+        mockWebServer.enqueue(
+                new MockResponse()
+                        .setResponseCode(200)
+                        .addHeader("Content-type", "application/json")
+                        .setBody(createBody()));
+    }
```

Comment:
```suggestion
    void setup() throws IOException {
        urlBuilder = new OpenWeatherUrlBuilder(generalSettings);
        yandexApi= new YandexApi(SKIP, LOCALHOST + mockWebServer.getPort(), SKIP);
        yandexService = new YandexWeatherService(restTemplate, generalSettings, urlBuilder);
    }
```
You can `@Mock` mockWebServer and generalSettings.
Then in code when you will need to get YandexApi just mock it by
`when(generalSettings.getYandexApi()).thenReturn(yandexApi);`